### PR TITLE
Slim 2105/external interface messages backend

### PR DESCRIPTION
**Pull request by coendamen**

### CODE REVIEW

File: osgp/shared/osgp-ws-smartmetering/pom.xml [hunk 1]
```diff
@@ -43,6 +43,24 @@
...
+            <configuration>
+              <bindingDirectory>src/main/resources/schemas</bindingDirectory>
+              <bindingFiles>sm-external-interface.xjb</bindingFiles>
+              <schemaDirectory>src/main/resources/schemas</schemaDirectory>
+              <xmlschema>true</xmlschema>
+              <schemaFiles>sm-external-interface.xsd</schemaFiles>
+              <!--              <schemaFiles>sm-external-interface.xsd</schemaFiles>-->
+              <staleFile>${project.build.directory}/jaxb2/.schema-smart-metering-external-stale-flag</staleFile>
+              <!-- USE THIS ONLY FOR THE FIRST EXECUTION -->
+              <clearOutputDir>false</clearOutputDir>
```
Comment (akoster) [inline osgp/shared/osgp-ws-smartmetering/pom.xml]: I think clearOutputDir can be removed. The default is true which clears the output directory before generating files. I think that's always ok.

File: osgp/shared/osgp-ws-smartmetering/pom.xml [hunk 2]
```diff
@@ -43,6 +43,24 @@
...
+            <goals>
+              <goal>xjc</goal>
+            </goals>
+            <configuration>
+              <bindingDirectory>src/main/resources/schemas</bindingDirectory>
+              <bindingFiles>sm-external-interface.xjb</bindingFiles>
+              <schemaDirectory>src/main/resources/schemas</schemaDirectory>
+              <xmlschema>true</xmlschema>
+              <schemaFiles>sm-external-interface.xsd</schemaFiles>
+              <!--              <schemaFiles>sm-external-interface.xsd</schemaFiles>-->
```
Comment (akoster) [inline osgp/shared/osgp-ws-smartmetering/pom.xml]: This comment can be removed

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-external-interface.xjb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<jxb:bindings version="2.0" xmlns:jxb="http://java.sun.com/xml/ns/jaxb"
```
Comment (akoster) [inline osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-external-interface.xjb]: There should be a copyright notice on top of this file. See the other binding files.

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-external-interface.xjb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+   </jxb:bindings>
+   <jxb:bindings schemaLocation="sm-bundle.xsd">
+      <jxb:schemaBindings>
+         <jxb:package name="org.opensmartgridplatform.adapter.ws.schema.smartmetering.bundle" />
+      </jxb:schemaBindings>
+      <jxb:bindings node="/xs:schema/xs:complexType[@name='Actions']/xs:sequence/xs:choice">
+         <jxb:property name="ActionList" />
+      </jxb:bindings>
+      <jxb:bindings node="/xs:schema/xs:complexType[@name='AllResponses']/xs:sequence/xs:choice">
+         <jxb:property name="ResponseList" />
```
Comment (akoster) [inline osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-external-interface.xjb]: Why do we need to rename Actions and AllResponses to ActionList and ResponseList? If it's not necessary I suggest either removing this customization or renaming the fields in the XSD's

File: osgp/shared/osgp-ws-smartmetering/pom.xml (hunk 2 above)
Comment (coendamen) [reply osgp/shared/osgp-ws-smartmetering/pom.xml]: Done

File: osgp/shared/osgp-ws-smartmetering/pom.xml (hunk 1 above)
Comment (coendamen) [reply osgp/shared/osgp-ws-smartmetering/pom.xml]: ik heb het uitgeprobeerd, dan blijven er maar twee packages over... dus ik laat het even zo.

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-external-interface.xjb (hunk 3 above)
Comment (coendamen) [reply osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-external-interface.xjb]: Done

**Pull request merged**